src/setup/node.rs

```diff
@@ -43,37 +43,69 @@ pub enum ChildExitCode {
     ErrorCode(Option<i32>),
 }
 
+/// Node type is used to select different startup configurations.
+pub enum NodeType {
+    /// A non-validator node without any preloaded data.
+    Stateless,
+    /// A validator node with a preloaded ledger data.
+    Stateful
+}
+
 pub struct NodeBuilder {
     conf: NodeConfig,
     meta: NodeMetaData,
 }
 
 impl NodeBuilder {
     /// Creates new [NodeBuilder]. Initial state is taken from `state` path, the node will run in `target` path.
-    pub fn new(state: Option<PathBuf>, target: PathBuf) -> Result<Self> {
-        if !target.exists() {
-            fs::create_dir_all(&target)?;
-        }
-        let mut copy_options = CopyOptions::new();
-        copy_options.content_only = true;
-        copy_options.overwrite = true;
+    pub fn new() -> Result<Self> {
         let setup_path = build_ripple_work_path()?.join(RIPPLE_SETUP_DIR);
-        copy(&setup_path, &target, &copy_options)?;
-        if let Some(source) = state {
-            copy(&source, &target, &copy_options)?;
-        }
-        let conf = NodeConfig::new(target);
-        let meta = NodeMetaData::new(conf.path.clone())?;
+
+        let conf = NodeConfig::new();
+        let meta = NodeMetaData::new(setup_path)?;
+
```

Comment:
Small note here:

Metadata shouldn't really care about the TempDir target directory.
Metadata should only have a fixed dependency towards .ziggurat/ripple/setup/config.toml.